### PR TITLE
Feature/ink 4.0.0

### PR DESCRIPTION
**Pull request by boyswan**

### CODE REVIEW

**Comment by Maar-io** [comment]
the CI is failing. Make update to github workflow. Example [here](https://github.com/swanky-dapps/nft/blob/main/.github/workflows/test.yml)

File: crates/common/src/types.rs [hunk 1]
```diff
@@ -26,10 +22,10 @@ pub type AssetId = u32;
 pub type EquippableGroupId = u32;
 
 /// Part's details
-#[derive(scale::Encode, scale::Decode, SpreadLayout, PackedLayout, Default, Debug)]
+#[derive(scale::Encode, scale::Decode, Default, Debug)]
 #[cfg_attr(
     feature = "std",
-    derive(scale_info::TypeInfo, ink_storage::traits::StorageLayout)
+    derive(scale_info::TypeInfo, ink::storage::traits::StorageLayout)
```
Comment (Maar-io) [inline crates/common/src/types.rs:28]: is `StorageLayout` needed?

File: crates/equippable/src/traits.rs [hunk 2]
```diff
@@ -3,7 +3,7 @@ use rmrk_common::{
     errors::Result,
     types::*,
 };
-// use ink_prelude::vec::Vec;
+// use ink::prelude::vec::Vec;
```
Comment (Maar-io) [inline crates/equippable/src/traits.rs:6]: delete?

File: examples/equippable/Cargo.toml [hunk 3]
```diff
@@ -5,20 +5,14 @@ authors = ["Stake Technologies <devops@stake.co.jp>"]
...
-
-openbrush = { tag = "v2.3.0", git = "https://github.com/Supercolony-net/openbrush-contracts", default-features = false, features = ["access_control", "psp34", "reentrancy_guard"] }
+scale-info = { version = "2.3", default-features = false, features = ["derive"], optional = true }
+openbrush = { tag = "3.0.0", git = "https://github.com/727-Ventures/openbrush-contracts", default-features = false, features = ["access_control", "reentrancy_guard", "psp34"] }
 rmrk = { path = "../../crates/rmrk", default-features = false  }
 
 
+
 [lib]
 name = "rmrk_example_equippable"
```
Comment (Maar-io) [inline examples/equippable/Cargo.toml:17]: `name` is not needed any more

File: examples/mintable/Cargo.toml [hunk 4]
```diff
@@ -5,20 +5,12 @@ authors = ["Stake Technologies <devops@stake.co.jp>"]
...
-scale-info = { version = "2", default-features = false, features = ["derive"], optional = true }
-
-openbrush = { tag = "v2.3.0", git = "https://github.com/Supercolony-net/openbrush-contracts", default-features = false, features = ["access_control", "psp34", "reentrancy_guard"] }
+scale-info = { version = "2.3", default-features = false, features = ["derive"], optional = true }
+openbrush = { tag = "3.0.0", git = "https://github.com/727-Ventures/openbrush-contracts", default-features = false, features = ["access_control", "reentrancy_guard", "psp34"] }
 rmrk = { path = "../../crates/rmrk", default-features = false  }
 
-
 [lib]
 name = "rmrk_example_mintable"
```
Comment (Maar-io) [inline examples/mintable/Cargo.toml:15]: remove name

**Review by Maar-io — APPROVED** [review]
LGTM

File: crates/common/src/types.rs (hunk 1 above)
Comment (boyswan) [reply crates/common/src/types.rs:28]: Yeah we still need storage layout for the types - fails to compile without them.

**Pull request merged**